### PR TITLE
Fix crowbar_register for s390x

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

**Comment by toabctl** [comment]
+1

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -672,16 +672,16 @@ function add_sdk_repo()
 {
     case "$cloudsource" in
         develcloud6|GM6|GM6+up)
-            zypper ar -f http://$susedownload/update/build.suse.de/SUSE/Products/SLE-SDK/12-SP1/x86_64/product/ SDK-SP1
-            zypper ar -f http://$susedownload/update/build.suse.de/SUSE/Updates/SLE-SDK/12-SP1/x86_64/update/ SDK-SP1-Update
+            zypper ar -f http://$susedownload/update/build.suse.de/SUSE/Products/SLE-SDK/12-SP1/$(uname -m)/product/ SDK-SP1
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: does it make sense to have this in a variable globally?

Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: you mean the output of uname -m ?

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -2030,7 +2030,7 @@ function onadmin_allocate()
         curl http://$clouddata/git/automation/scripts/qa1_nodes_reboot | bash
     fi
 
-    wait_for 50 10 'test $(get_all_discovered_nodes | wc -l) -ge 1' "first node to be discovered"
+    [[ $nodenumber -gt 0 ]] && wait_for 50 10 'test $(get_all_discovered_nodes | wc -l) -ge 1' "first node to be discovered"
     wait_for 100 10 '[[ $(get_all_discovered_nodes | wc -l) -ge $nodenumber ]]' "all nodes to be discovered"
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: if the nodenumber is 0 then the wait for all nodes to be discovered should be skipped as well, right?

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: yea, sounds like a good idea

File: scripts/mkcloud [hunk 3]
```diff
@@ -474,6 +474,11 @@ function setuphost()
             libvirt $kvmpkg $osloader $ipxe lvm2 curl wget bridge-utils \
             dnsmasq netcat-openbsd ebtables libvirt-python
         [ "$?" == 0 -o "$?" == 4 ] || complain 10 "setuphost failed to install required packages"
+
+        # enable KVM
+        [[ $(uname -m) = s390x ]] && {
+            echo 1 > /proc/sys/vm/allocate_pgste
+        }
```
Comment (jdsn) [inline scripts/mkcloud]: As we have several places with such checks, maybe it makes sense to have a `is_arch` function.
But not necessarily as part of this PR. Could be refactored later.

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: well, we already have it. Just use `$arch`

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: it passes immediately since get_all_discovered_nodes returns  0, which is -ge 0. so no change needed.

**Comment by jdsn** [comment]
:+1: if gating passes

**Comment by jdsn** [comment]
@dirkmueller needs rebase ("automatic merge failed")

**Comment by dirkmueller** [comment]
gating failure is unrelated so far.

**Comment by toabctl** [comment]
+1

**Pull request merged**